### ISSUE TITLE
Anuglar 9 SSR support

### ISSUE THREAD

**Issue by kamilogerto2** [hodgef/simple-keyboard#520]
Hey. :)
It seems that there is a problem with window reference when running the project in SSR mode with linked simple-keyboard. Since SSR is very sensitive for window checking - it's possible to improve the library with that?

For now it's one of the blockers for us to switch fully to Angular 9.

![Zrzut ekranu z 2020-04-20 16-31-25](https://user-images.githubusercontent.com/15246741/79763888-ed35aa80-8324-11ea-8402-9c50c211f8df.png)

**Comment by hodgef** [comment]
Hey @kamilogerto2,

For SSR, you can use a conditional import

```
if (window) {
  import("simple-keyboard").then(KeyboardClass => {
    const Keyboard = KeyboardClass.default

    this.keyboard = new Keyboard({
      onChange: this.onChange,
      onKeyPress: this.onKeyPress,
    });
  });
}
```

Best,
Francisco Hodge

**Comment by kamilogerto2** [comment]
@hodgef I was thinking about that but also I know that the libraries are trying to be prepared for such cases. It's easier for developers later. But - thank you for involvement ;)

**Comment by hodgef** [comment]
I see. It's a fair request, I will add it to the [Enhancements project](https://github.com/hodgef/simple-keyboard/projects/3) and hopefully take it up soon.

Thanks,
Francisco Hodge

**Comment by kamilogerto2** [comment]
Thanks! :)

**Comment by kamilogerto2** [comment]
@hodgef I just implemented the dynamically import but then due to dynamic - our animation is a little bit crash. So for now I will leave it like it is but I am waiting for that improvement from your side (unfortunately I don't have a lot of time for doing that :( ) ;)

**Comment by hodgef** [comment]
Hello @kamilogerto2,

Unfortunately I won't be able to pick this one up in the near future. It will be released eventually but I don't have an estimate.

Thanks,
Francisco Hodge

**Comment by kamilogerto2** [comment]
Just a quick reference how it was solved in another project :) https://github.com/naver/hammer.js/commit/af3f98f663ad9d046e496d1c7cf81f0ff9c74660

**Comment by hodgef** [comment]
Fixed in v3.0.0. Thanks!

**Comment by Romko775** [comment]
@hodgef Hi! Having same issue with Angular 11 SSR & self

```
ReferenceError: self is not defined
    at Object.BsoO (...\server\main.js:1:407835)

```


```
BsoO:function(module,exports,__webpack_require__){self,module.exports=function(){"use strict";var t={d:function(a,e){for(var s in e)t.o(e,s)&&!t.o(a,s)&&Object.defineProperty(a,s,{enumerable:!0,get:e[s]})},o:function(t,a){return Object.prototype.hasOwnProperty.call(t,a)},r:function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})}},a={};t.r(a),t.d(a,{default:function(){return E}});var e={layout:{default:["\u0630 1 2 3 4 5 6 7 8 9 0 - = {bksp}","{tab} \u0636 \u0635 \u062b \u0642 ...
```
